pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -127,24 +126,39 @@ func ctrStop(containerID string, force bool) error {
 	return nil
 }
 
-// ctrRm remove an existing container
-func ctrRm(containerID string) error {
+// ctrRm remove an existing container. If silent is true, then operation failures are ignored and no error is returned
+func ctrRm(containerID string, silent bool) error {
 	log.Infof("ctrRm %s\n", containerID)
 
 	container, err := loadContainer(containerID)
 	if err != nil {
-		return err
+		log.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		if !silent {
+			return fmt.Errorf("ctrRm: exception while loading container: %v. %v", containerID, err.Error())
+		}
```

Comment:
Can avoid duplication in all these cases if you do
 err = fmt.Errorf(...)
 log.Error(err)
 if !silent {
    return err
}


Also, in the silent case shouldn't you return nil? How do we know it is safe to drive on when a previous command failed?>

Reply:
> Also, in the silent case shouldn't you return nil?
This method returns nil in silent case at the end

>How do we know it is safe to drive on when a previous command failed?
There were cases where the clean-up was half done on device reboot (caused by this issue: https://www.pivotaltracker.com/story/show/171802157) because of which in some cases the container was deleted, rootfs inside bundle was unmounted but the bundle was not deleted. So the purpose I added silent mode was to delete/clean-up the things that are present and ignore the rest.
Clean-up in silent mode is only triggered in 2 cases
1) After we create a container if there is an error on some intermittent step while creating a bundle for the container. 
2) After device reboot, when we restart the AppInstance

Reply:
>Can avoid duplication in all these cases if you do

Done.